Caution: the Python 3.7 module is intended for use on Ubuntu 18.04 machines, but most packages will probably work fine on Ubuntu 20.04.

Loading tensorflow/1.15.0-gpu-py37
  Loading requirement: python/3.7
No protocol specified
WARNING:tensorflow:
The TensorFlow contrib module will not be included in TensorFlow 2.0.
For more information, please see:
  * https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md
  * https://github.com/tensorflow/addons
  * https://github.com/tensorflow/io (for I/O related ops)
If you depend on functionality not listed there, please file an issue.

WARNING:tensorflow:
The TensorFlow contrib module will not be included in TensorFlow 2.0.
For more information, please see:
  * https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md
  * https://github.com/tensorflow/addons
  * https://github.com/tensorflow/io (for I/O related ops)
If you depend on functionality not listed there, please file an issue.

hparams:
hparams:
  batch_size: 16
  debug_normalize: 0
  labels:
    num_train: 46000
    source: https://openaipublic.blob.core.windows.net/lm-human-preferences/labels/tldr/online_45k.json
    type: best_of_4
  lr: 2.5e-05
  normalize_after: True
  normalize_before: True
  normalize_samples: 256
  rollout_batch_size: 128
  run:
    log_interval: 10
    save_dir: /tmp/save/train_reward/test-2202152050
    save_interval: 50
    seed: 1
  task:
    end_text: None
    penalty_reward_value: -1
    policy:
      initial_model: 124M
      temperature: 0.7
    query_dataset: tldr
    query_length: 500
    query_prefix: 
    query_suffix: 

TL;DR:
    response_length: 75
    start_text: None
  batch_size: 16
  debug_normalize: 0
  labels:
    num_train: 46000
    source: https://openaipublic.blob.core.windows.net/lm-human-preferences/labels/tldr/online_45k.json
    type: best_of_4
  lr: 2.5e-05
  normalize_after: True
  normalize_before: True
  normalize_samples: 256
  rollout_batch_size: 128
  run:
    log_interval: 10
    save_dir: /tmp/save/train_reward/test-2202152050
    save_interval: 50
    seed: 1
  task:
    end_text: None
    penalty_reward_value: -1
    policy:
      initial_model: 124M
      temperature: 0.7
    query_dataset: tldr
    query_length: 500
    query_prefix: 
    query_suffix: 

TL;DR:
    response_length: 75
    start_text: None
    truncate_after: 55
    truncate_token: 198
    truncate_after: 55
    truncate_token: 198
WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:116: The name tf.set_random_seed is deprecated. Please use tf.compat.v1.set_random_seed instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/encodings.py:133: The name tf.gfile.Open is deprecated. Please use tf.io.gfile.GFile instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:116: The name tf.set_random_seed is deprecated. Please use tf.compat.v1.set_random_seed instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/encodings.py:133: The name tf.gfile.Open is deprecated. Please use tf.io.gfile.GFile instead.

model_hparams:
  attn_pdrop: 0.1
  embd_pdrop: 0.1
  head_pdrop: 0.1
  n_ctx: 1024
  n_embd: 768
  n_head: 12
  n_layer: 12
  n_vocab: 50257
  resid_pdrop: 0.1
model_hparams:
  attn_pdrop: 0.1
  embd_pdrop: 0.1
  head_pdrop: 0.1
  n_ctx: 1024
  n_embd: 768
  n_head: 12
  n_layer: 12
  n_vocab: 50257
  resid_pdrop: 0.1
WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/policy.py:24: The name tf.variable_scope is deprecated. Please use tf.compat.v1.variable_scope instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:550: The name tf.placeholder is deprecated. Please use tf.compat.v1.placeholder instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/policy.py:24: The name tf.variable_scope is deprecated. Please use tf.compat.v1.variable_scope instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:550: The name tf.placeholder is deprecated. Please use tf.compat.v1.placeholder instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/model.py:297: where (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where
WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/model.py:297: where (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where
WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/model.py:309: The name tf.get_variable is deprecated. Please use tf.compat.v1.get_variable instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/model.py:309: The name tf.get_variable is deprecated. Please use tf.compat.v1.get_variable instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/model.py:71: The name tf.rsqrt is deprecated. Please use tf.math.rsqrt instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/model.py:71: The name tf.rsqrt is deprecated. Please use tf.math.rsqrt instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:172: The name tf.matrix_band_part is deprecated. Please use tf.linalg.band_part instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/model.py:339: The name tf.assert_none_equal is deprecated. Please use tf.compat.v1.assert_none_equal instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:172: The name tf.matrix_band_part is deprecated. Please use tf.linalg.band_part instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/model.py:339: The name tf.assert_none_equal is deprecated. Please use tf.compat.v1.assert_none_equal instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:531: The name tf.trainable_variables is deprecated. Please use tf.compat.v1.trainable_variables instead.

Param ref_policy/model/heads/value/w is missing from checkpoint ./gpt-2/models/124M/main/model.ckpt
Param ref_policy/model/heads/value/b is missing from checkpoint ./gpt-2/models/124M/main/model.ckpt
WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/trained_models.py:66: The name tf.train.init_from_checkpoint is deprecated. Please use tf.compat.v1.train.init_from_checkpoint instead.

WARNING:tensorflow:Entity <function make_query_sampler.<locals>.<lambda> at 0x7f6988765950> could not be transformed and will be executed as-is. Please report this to the AutoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: module 'gast' has no attribute 'Index'
WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/lm_tasks.py:101: DatasetV1.make_one_shot_iterator (from tensorflow.python.data.ops.dataset_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use `for ... in dataset:` to iterate over a dataset. If using `tf.estimator`, return the `Dataset` object directly from your input function. As a last resort, you can use `tf.compat.v1.data.make_one_shot_iterator(dataset)`.
WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py:284: The name tf.train.create_global_step is deprecated. Please use tf.compat.v1.train.create_global_step instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:217: The name tf.GraphKeys is deprecated. Please use tf.compat.v1.GraphKeys instead.

WARNING:tensorflow:From /usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/ops/resource_variable_ops.py:1630: calling BaseResourceVariable.__init__ (from tensorflow.python.ops.resource_variable_ops) with constraint is deprecated and will be removed in a future version.
Instructions for updating:
If using Keras pass *_constraint arguments to layers.
Param reward_model/model/heads/reward/w is missing from checkpoint ./gpt-2/models/124M/main/model.ckpt
Param reward_model/model/heads/reward/b is missing from checkpoint ./gpt-2/models/124M/main/model.ckpt
Param reward_model/reward_norm/gain is missing from checkpoint ./gpt-2/models/124M/main/model.ckpt
Param reward_model/reward_norm/bias is missing from checkpoint ./gpt-2/models/124M/main/model.ckpt
WARNING:tensorflow:Entity <function make_query_sampler.<locals>.<lambda> at 0x7fe3ea830950> could not be transformed and will be executed as-is. Please report this to the AutoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: module 'gast' has no attribute 'Index'
WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/lm_tasks.py:101: DatasetV1.make_one_shot_iterator (from tensorflow.python.data.ops.dataset_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use `for ... in dataset:` to iterate over a dataset. If using `tf.estimator`, return the `Dataset` object directly from your input function. As a last resort, you can use `tf.compat.v1.data.make_one_shot_iterator(dataset)`.
WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py:284: The name tf.train.create_global_step is deprecated. Please use tf.compat.v1.train.create_global_step instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:217: The name tf.GraphKeys is deprecated. Please use tf.compat.v1.GraphKeys instead.

WARNING:tensorflow:From /usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/ops/resource_variable_ops.py:1630: calling BaseResourceVariable.__init__ (from tensorflow.python.ops.resource_variable_ops) with constraint is deprecated and will be removed in a future version.
Instructions for updating:
If using Keras pass *_constraint arguments to layers.
WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:531: The name tf.trainable_variables is deprecated. Please use tf.compat.v1.trainable_variables instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:481: py_func (from tensorflow.python.ops.script_ops) is deprecated and will be removed in a future version.
Instructions for updating:
tf.py_func is deprecated in TF V2. Instead, there are two
    options available in V2.
    - tf.py_function takes a python function which manipulates tf eager
    tensors instead of numpy arrays. It's easy to convert a tf eager tensor to
    an ndarray (just call tensor.numpy()) but having access to eager tensors
    means `tf.py_function`s can use accelerators such as GPUs as well as
    being differentiable using a gradient tape.
    - tf.numpy_function maintains the semantics of the deprecated tf.py_func
    (it is not differentiable, and manipulates numpy arrays). It drops the
    stateful argument making all functions stateful.
    
WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:630: The name tf.train.AdamOptimizer is deprecated. Please use tf.compat.v1.train.AdamOptimizer instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:481: py_func (from tensorflow.python.ops.script_ops) is deprecated and will be removed in a future version.
Instructions for updating:
tf.py_func is deprecated in TF V2. Instead, there are two
    options available in V2.
    - tf.py_function takes a python function which manipulates tf eager
    tensors instead of numpy arrays. It's easy to convert a tf eager tensor to
    an ndarray (just call tensor.numpy()) but having access to eager tensors
    means `tf.py_function`s can use accelerators such as GPUs as well as
    being differentiable using a gradient tape.
    - tf.numpy_function maintains the semantics of the deprecated tf.py_func
    (it is not differentiable, and manipulates numpy arrays). It drops the
    stateful argument making all functions stateful.
    
WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:630: The name tf.train.AdamOptimizer is deprecated. Please use tf.compat.v1.train.AdamOptimizer instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py:297: The name tf.train.Saver is deprecated. Please use tf.compat.v1.train.Saver instead.

Will save to /tmp/save/train_reward/test-2202152050
rank 1: gpus = [0, 1], our gpu = 1
rank 0: gpus = [0, 1], our gpu = 0
WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py:314: The name tf.global_variables_initializer is deprecated. Please use tf.compat.v1.global_variables_initializer instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py:314: The name tf.global_variables_initializer is deprecated. Please use tf.compat.v1.global_variables_initializer instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py:315: The name tf.local_variables_initializer is deprecated. Please use tf.compat.v1.local_variables_initializer instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py:315: The name tf.local_variables_initializer is deprecated. Please use tf.compat.v1.local_variables_initializer instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py:322: The name tf.get_default_graph is deprecated. Please use tf.compat.v1.get_default_graph instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:108: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.

2022-02-16 00:13:05.242254: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 AVX512F FMA
WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py:322: The name tf.get_default_graph is deprecated. Please use tf.compat.v1.get_default_graph instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:108: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.

2022-02-16 00:13:05.243039: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 AVX512F FMA
2022-02-16 00:13:05.270908: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2200000000 Hz
2022-02-16 00:13:05.270938: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2200000000 Hz
2022-02-16 00:13:05.271283: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5619c45aa870 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2022-02-16 00:13:05.271311: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2022-02-16 00:13:05.271245: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x558d9b55fe20 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2022-02-16 00:13:05.271285: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2022-02-16 00:13:05.276380: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1
2022-02-16 00:13:05.276828: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1
2022-02-16 00:13:05.960825: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x558d9b37d500 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
2022-02-16 00:13:05.960849: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): NVIDIA A100 80GB PCIe, Compute Capability 8.0
2022-02-16 00:13:05.964082: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5619c2384500 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
2022-02-16 00:13:05.964100: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): NVIDIA A100 80GB PCIe, Compute Capability 8.0
2022-02-16 00:13:05.971692: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties: 
name: NVIDIA A100 80GB PCIe major: 8 minor: 0 memoryClockRate(GHz): 1.41
pciBusID: 0000:ce:00.0
2022-02-16 00:13:05.972146: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.0
2022-02-16 00:13:05.974887: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10.0
2022-02-16 00:13:05.976470: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10.0
2022-02-16 00:13:05.976913: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10.0
2022-02-16 00:13:05.978077: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10.0
2022-02-16 00:13:05.978518: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1618] Found device 0 with properties: 
name: NVIDIA A100 80GB PCIe major: 8 minor: 0 memoryClockRate(GHz): 1.41
pciBusID: 0000:d1:00.0
2022-02-16 00:13:05.978959: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.0
2022-02-16 00:13:05.979204: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10.0
2022-02-16 00:13:05.980195: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10.0
2022-02-16 00:13:05.981180: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10.0
2022-02-16 00:13:05.981447: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10.0
2022-02-16 00:13:05.982152: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2022-02-16 00:13:05.982855: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10.0
2022-02-16 00:13:05.983934: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10.0
2022-02-16 00:13:05.987396: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2022-02-16 00:13:05.988737: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2022-02-16 00:13:05.988783: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.0
2022-02-16 00:13:05.999651: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 1
2022-02-16 00:13:05.999688: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.0
2022-02-16 00:13:06.003079: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2022-02-16 00:13:06.003100: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0 
2022-02-16 00:13:06.003106: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N 
2022-02-16 00:13:06.012717: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2022-02-16 00:13:06.012735: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      1 
2022-02-16 00:13:06.012741: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 1:   N 
2022-02-16 00:13:06.016597: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 72895 MB memory) -> physical GPU (device: 0, name: NVIDIA A100 80GB PCIe, pci bus id: 0000:ce:00.0, compute capability: 8.0)
2022-02-16 00:13:06.024687: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 72895 MB memory) -> physical GPU (device: 1, name: NVIDIA A100 80GB PCIe, pci bus id: 0000:d1:00.0, compute capability: 8.0)
WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:568: The name tf.RunOptions is deprecated. Please use tf.compat.v1.RunOptions instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:570: The name tf.get_default_session is deprecated. Please use tf.compat.v1.get_default_session instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:568: The name tf.RunOptions is deprecated. Please use tf.compat.v1.RunOptions instead.

WARNING:tensorflow:From /accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py:570: The name tf.get_default_session is deprecated. Please use tf.compat.v1.get_default_session instead.

Num labels found in source: 46287
Num labels found in source: 46287
2022-02-16 00:16:25.364097: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10.0
2022-02-16 00:16:26.156440: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10.0
2022-02-16 00:21:27.348086: E tensorflow/stream_executor/cuda/cuda_blas.cc:428] failed to run cuBLAS routine: CUBLAS_STATUS_EXECUTION_FAILED
Traceback (most recent call last):
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/client/session.py", line 1365, in _do_call
    return fn(*args)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/client/session.py", line 1350, in _run_fn
    target_list, run_metadata)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/client/session.py", line 1443, in _call_tf_sessionrun
    run_metadata)
tensorflow.python.framework.errors_impl.InternalError: 2 root error(s) found.
  (0) Internal: Blas GEMM launch failed : a.shape=(32384, 768), b.shape=(768, 2304), m=32384, n=2304, k=768
	 [[{{node sample_seq/step/h0/attn/c_attn/MatMul}}]]
	 [[sample_seq/while/Identity/_716]]
  (1) Internal: Blas GEMM launch failed : a.shape=(32384, 768), b.shape=(768, 2304), m=32384, n=2304, k=768
	 [[{{node sample_seq/step/h0/attn/c_attn/MatMul}}]]
0 successful operations.
0 derived errors ignored.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py", line 328, in train
    reward_trainer.train()
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py", line 236, in train
    self.normalize(self.sample_policy_responses, target_mean, target_std)
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py", line 214, in normalize
    query_responses = sample_fn(self.hparams.normalize_samples)
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py", line 201, in sample_policy_responses
    return [sample_policy_batch() for _ in range(n_batches)]
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py", line 201, in <listcomp>
    return [sample_policy_batch() for _ in range(n_batches)]
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py", line 570, in run
    return tf.get_default_session().run(op, feed_dict=feed, options=run_options, run_metadata=None)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/client/session.py", line 956, in run
    run_metadata_ptr)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/client/session.py", line 1180, in _run
    feed_dict_tensor, options, run_metadata)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/client/session.py", line 1359, in _do_run
    run_metadata)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/client/session.py", line 1384, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors_impl.InternalError: 2 root error(s) found.
  (0) Internal: Blas GEMM launch failed : a.shape=(32384, 768), b.shape=(768, 2304), m=32384, n=2304, k=768
	 [[node sample_seq/step/h0/attn/c_attn/MatMul (defined at /usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/framework/ops.py:1748) ]]
	 [[sample_seq/while/Identity/_716]]
  (1) Internal: Blas GEMM launch failed : a.shape=(32384, 768), b.shape=(768, 2304), m=32384, n=2304, k=768
	 [[node sample_seq/step/h0/attn/c_attn/MatMul (defined at /usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/framework/ops.py:1748) ]]
0 successful operations.
0 derived errors ignored.

Original stack trace for 'sample_seq/step/h0/attn/c_attn/MatMul':
  File "<string>", line 1, in <module>
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py", line 291, in train
    comm=comm,
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py", line 192, in __init__
    @utils.graph_function()
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py", line 552, in decorate
    op = make_op(**phs)
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py", line 196, in sample_policy_batch
    queries=queries, length=hparams.task.response_length)['responses']
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/policy.py", line 106, in respond_op
    extra_outputs={'values':tf.float32},
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/sample.py", line 26, in sample_sequence
    context_output = step(model_hparams, context)
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/policy.py", line 56, in step_core
    do_dropout=do_dropout, padding_token=self.padding_token)
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/model.py", line 330, in __call__
    hparams=self.hparams, seed=block_seed)
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/model.py", line 200, in block
    'attn', nx, past=past, mask=mask, do_dropout=do_dropout, scale=scale, hparams=hparams, seed=attn_seed)
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/model.py", line 166, in attn
    c = conv1x1(x, 'c_attn', n_state * 3, w_init_stdev=w_init_stdev)
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/model.py", line 96, in conv1x1
    c = tf.matmul(tf.reshape(x, [-1, nx]), w) + b
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/util/dispatch.py", line 180, in wrapper
    return target(*args, **kwargs)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/ops/math_ops.py", line 2754, in matmul
    a, b, transpose_a=transpose_a, transpose_b=transpose_b, name=name)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/ops/gen_math_ops.py", line 6136, in mat_mul
    name=name)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/framework/op_def_library.py", line 794, in _apply_op_helper
    op_def=op_def)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/util/deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/framework/ops.py", line 3357, in create_op
    attrs, op_def, compute_device)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/framework/ops.py", line 3426, in _create_op_internal
    op_def=op_def)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/framework/ops.py", line 1748, in __init__
    self._traceback = tf_stack.extract_stack()

2022-02-16 00:21:47.511350: E tensorflow/stream_executor/cuda/cuda_blas.cc:428] failed to run cuBLAS routine: CUBLAS_STATUS_EXECUTION_FAILED
Traceback (most recent call last):
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/client/session.py", line 1365, in _do_call
    return fn(*args)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/client/session.py", line 1350, in _run_fn
    target_list, run_metadata)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/client/session.py", line 1443, in _call_tf_sessionrun
    run_metadata)
tensorflow.python.framework.errors_impl.InternalError: 2 root error(s) found.
  (0) Internal: Blas GEMM launch failed : a.shape=(32384, 768), b.shape=(768, 2304), m=32384, n=2304, k=768
	 [[{{node sample_seq/step/h0/attn/c_attn/MatMul}}]]
	 [[strided_slice_12/_1397]]
  (1) Internal: Blas GEMM launch failed : a.shape=(32384, 768), b.shape=(768, 2304), m=32384, n=2304, k=768
	 [[{{node sample_seq/step/h0/attn/c_attn/MatMul}}]]
0 successful operations.
0 derived errors ignored.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py", line 328, in train
    reward_trainer.train()
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py", line 236, in train
    self.normalize(self.sample_policy_responses, target_mean, target_std)
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py", line 214, in normalize
    query_responses = sample_fn(self.hparams.normalize_samples)
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py", line 201, in sample_policy_responses
    return [sample_policy_batch() for _ in range(n_batches)]
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py", line 201, in <listcomp>
    return [sample_policy_batch() for _ in range(n_batches)]
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py", line 570, in run
    return tf.get_default_session().run(op, feed_dict=feed, options=run_options, run_metadata=None)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/client/session.py", line 956, in run
    run_metadata_ptr)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/client/session.py", line 1180, in _run
    feed_dict_tensor, options, run_metadata)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/client/session.py", line 1359, in _do_run
    run_metadata)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/client/session.py", line 1384, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors_impl.InternalError: 2 root error(s) found.
  (0) Internal: Blas GEMM launch failed : a.shape=(32384, 768), b.shape=(768, 2304), m=32384, n=2304, k=768
	 [[node sample_seq/step/h0/attn/c_attn/MatMul (defined at /usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/framework/ops.py:1748) ]]
	 [[strided_slice_12/_1397]]
  (1) Internal: Blas GEMM launch failed : a.shape=(32384, 768), b.shape=(768, 2304), m=32384, n=2304, k=768
	 [[node sample_seq/step/h0/attn/c_attn/MatMul (defined at /usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/framework/ops.py:1748) ]]
0 successful operations.
0 derived errors ignored.

Original stack trace for 'sample_seq/step/h0/attn/c_attn/MatMul':
  File "<string>", line 1, in <module>
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py", line 291, in train
    comm=comm,
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py", line 192, in __init__
    @utils.graph_function()
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/utils/core.py", line 552, in decorate
    op = make_op(**phs)
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/train_reward.py", line 196, in sample_policy_batch
    queries=queries, length=hparams.task.response_length)['responses']
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/policy.py", line 106, in respond_op
    extra_outputs={'values':tf.float32},
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/sample.py", line 26, in sample_sequence
    context_output = step(model_hparams, context)
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/policy.py", line 56, in step_core
    do_dropout=do_dropout, padding_token=self.padding_token)
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/model.py", line 330, in __call__
    hparams=self.hparams, seed=block_seed)
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/model.py", line 200, in block
    'attn', nx, past=past, mask=mask, do_dropout=do_dropout, scale=scale, hparams=hparams, seed=attn_seed)
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/model.py", line 166, in attn
    c = conv1x1(x, 'c_attn', n_state * 3, w_init_stdev=w_init_stdev)
  File "/accounts/projects/jsteinhardt/aypan/lm-human-preferences/lm_human_preferences/language/model.py", line 96, in conv1x1
    c = tf.matmul(tf.reshape(x, [-1, nx]), w) + b
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/util/dispatch.py", line 180, in wrapper
    return target(*args, **kwargs)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/ops/math_ops.py", line 2754, in matmul
    a, b, transpose_a=transpose_a, transpose_b=transpose_b, name=name)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/ops/gen_math_ops.py", line 6136, in mat_mul
    name=name)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/framework/op_def_library.py", line 794, in _apply_op_helper
    op_def=op_def)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/util/deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/framework/ops.py", line 3357, in create_op
    attrs, op_def, compute_device)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/framework/ops.py", line 3426, in _create_op_internal
    op_def=op_def)
  File "/usr/local/linux/anaconda3.7-extra-tf1.15.0/lib/python3.7/site-packages/tensorflow_core/python/framework/ops.py", line 1748, in __init__
    self._traceback = tf_stack.extract_stack()

--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpiexec detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[16048,1],1]
  Exit code:    1
--------------------------------------------------------------------------
